packages/algorand-js/test/src/interpreter/opcode-list.ts

```diff
@@ -412,4 +415,445 @@ describe("Teal Opcodes", function () {
+
+    it("should push 1 to stack", () => {
+      const op = new LessThanEqualTo();
+      stack.push(BigInt('20'));
+      stack.push(BigInt('20'));
+
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt('1'));
+    });
+
+    it("should push 0 to stack", () => {
+      const op = new LessThanEqualTo();
+      stack.push(BigInt('50'));
+      stack.push(BigInt('100'));
+
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt('0'));
+    });
+  });
+
+  describe("GreaterThanEqualTo", function () {
+    const stack = new Stack<StackElem>();
+
+    it("should push 1 to stack", () => {
+      const op = new GreaterThanEqualTo();
+      stack.push(BigInt('20'));
+      stack.push(BigInt('20'));
+
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt('1'));
+    });
+
+    it("should push 0 to stack", () => {
+      const op = new GreaterThanEqualTo();
+      stack.push(BigInt('500'));
+      stack.push(BigInt('100'));
+
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(top, BigInt('0'));
+    });
+  });
+
+  describe("And", () => {
+    const stack = new Stack<StackElem>();
+
+    it("should push true to stack as both values are 1", () => {
+      stack.push(BigInt('1'));
+      stack.push(BigInt('1'));
+
+      const op = new And();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(BigInt('1'), top);
+    });
+
+    it("should push false to stack as one value is 0", () => {
+      stack.push(BigInt('0'));
+      stack.push(BigInt('1'));
+
+      const op = new And();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(BigInt('0'), top);
+    });
+  });
+
+  describe("Or", () => {
+    const stack = new Stack<StackElem>();
+
+    it("should push true to stack as one value is 1", () => {
+      stack.push(BigInt('0'));
+      stack.push(BigInt('1'));
+
+      const op = new Or();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(BigInt('1'), top);
+    });
+
+    it("should push false to stack as both values are 0", () => {
+      stack.push(BigInt('0'));
+      stack.push(BigInt('0'));
+
+      const op = new Or();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(BigInt('0'), top);
+    });
+  });
+
+  describe("EqualTo", () => {
+    const stack = new Stack<StackElem>();
+
+    it("should push true to stack", () => {
+      stack.push(BigInt('22'));
+      stack.push(BigInt('22'));
+
+      const op = new EqualTo();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(BigInt('1'), top);
+    });
+
+    it("should push false to stack", () => {
+      stack.push(BigInt('22'));
+      stack.push(BigInt('1'));
+
+      const op = new EqualTo();
+      op.execute(stack);
+
+      const top = stack.pop();
+      assert.equal(BigInt('0'), top);
+    });
+  });
```

Comment:
Please add tests for `bytes[]` comparison as well. Also for mixed types as well.
Eg - test for (`==`, `!=`):  bytes("1") == BigInt('1').